tests/scripts/all.sh

```diff
@@ -477,8 +476,9 @@ pre_check_git () {
 }
 
 pre_setup_keep_going () {
-    failure_summary=
-    failure_count=0
+    failure_count=0 # Number of failed components
+    last_failure_status=0 # Last failure status in this component
```

Comment:
Note: same for other initialisations/unset introduced in later commits.

Reply:
This part is only executed in keep-going mode.

Reply:
Yes, no question about that, my question was more about the fact that this part is executed once per run of `all.sh` rather than once per component, while it looked to me like this variable should be reset to 0 before running each component.

In the meantime I realised that the variable is only ever modified in a subshell, so the top-level shell doesn't need to reset it between components. I'm not sure if that was just me being slow or if it's worth adding a comment about that.